irohad/multi_sig_transactions/impl/mst_processor_impl.cpp

```diff
@@ -56,7 +63,12 @@ namespace iroha {
 
   auto FairMstProcessor::propagateBatchImpl(const iroha::DataType &batch)
       -> decltype(propagateBatch(batch)) {
-    shareState(storage_->updateOwnState(batch), batches_subject_);
+    auto state_update = storage_->updateOwnState(batch);
```

Comment:
I don't see a reason to introduce pair with bool response here. Method isEmpty is already implemented in MstState. 
Also, the current interface of updateOwnState is totally misleading - for one situation it returns one value, for another situation - another value.

UpdateOwnState possibly may return pair of two states: state with completed transactions and state with new signatures.

Reply:
AFAIK the second parameter reflects the state of a batch - completed or just updated. 
Nevertheless, the current implementation with std::pair is unclear and misleading.
My vote for the simplicity.